app/services/container_topology_service.rb

```diff
@@ -87,26 +83,10 @@ def entity_display_type(entity)
     end
   end
 
-  def entity_id(entity)
-    if entity.kind_of?(ManageIQ::Providers::ContainerManager)
-      id = entity.id.to_s
-    elsif entity.kind_of?(Host) || entity.kind_of?(Vm)
-      id = entity.uid_ems
-    else
-      id = entity.ems_ref
-    end
-    id
-  end
-
   def build_entity_data(entity)
-    type = entity_type(entity)
-    status = entity_status(entity)
-    data = {:id           => entity_id(entity),
-            :name         => entity.name,
-            :status       => status,
-            :kind         => type,
-            :display_kind => entity_display_type(entity),
-            :miq_id       => entity.id}
+    data = build_base_entity_data(entity)
+    data.merge!(:status       => entity_status(entity))
+    data.merge!(:display_kind => entity_display_type(entity))
```

Comment:
This can be a single merge!

```ruby
data.merge!(
  :status       => entity_status(entity)
  :display_kind => entity_display_type(entity)
)
```